### DISCUSSION TITLE
Is there support the execution of tasks in a serial manner, where instances of the same task are not allowed to run in parallel?

### DISCUSSION THREAD

**Discussion by HalavicH** [kagkarlsson/db-scheduler#390] [Q&A] [answered]
Hello, kagkarlsson and community!

I have a question regarding the capabilities of the library.
I would like to know if it is possible to **define tasks** in such a way that they are **not eligible for parallel execution** with other instances of the same task.

To clarify, let's say we have a task named **"TaskA"** that is currently **running**. If a user initiates the execution of "TaskA" **once again** (with a different instance) while the first instance is still running, can we ensure that the **second execution remains in a pending** state until the first execution is completed? Essentially, I'm looking for behavior similar to having a single executor thread dedicated to specific tasks.

If anyone has insights on whether this functionality is already available in the db-scheduler library or any suggestions on achieving this behavior, please share your thoughts. Thank you!

**Comment by kagkarlsson** [comment, marked as answer]
This is a feature that has been requested before but currently not supported. I don't think this is very easy to implement support for either. You might be able to work around it using a recurring task and some external table for queuing up the instances. The task would simply execute all instances available.

**Reply by HalavicH** [reply]
Hello again!

Thank you for your previous response. While exploring alternative approaches to implementing the desired feature, I came across the db-scheduler extension. It utilizes the StatsRegistry from your library to enable task logging. You can find more information about it here: [link to the db-scheduler-log GitHub repository](https://github.com/rocketbase-io/db-scheduler-log).

Now, I'm curious if there are any similar hooks within db-scheduler library specifically designed for task scheduling and execution. If such hooks exist, it might be possible to utilize them to achieve the goal of serial task execution.

**Reply by kagkarlsson** [reply]
None that I can think of. There are currently no possibility for such gates in db-scheduler to enable this in a reliable and efficient way.

You might be able to do a check for a running execution first-thing in your `ExecutionHandler` using the `SchedulerClient`, but it will not be perfect. Something like: 

https://github.com/kagkarlsson/db-scheduler/blob/poc_serial/examples/features/src/main/java/com/github/kagkarlsson/examples/SerialMain.java

**Reply by HalavicH** [reply]
Thanks again for your quick response and example.

I've also thought about the approach you've provided as a last resort.
Anyway, this might work for me.The only downside for me is task log pollution with tasks that wait for their turn
